packages/core/core/src/services/document-service/transform/fields.ts

```diff
@@ -0,0 +1,3 @@
+export const transformFields = (fields: string[]): string[] => {
+  return fields.map((field) => (field === 'id' ? 'documentId' : field));
```

Comment:
could we make sure the document id is always present in fields?

Meaning, if the `fields` field is present, we need to be sure the documentId is always present, so the documentID -> id switch can be made on every document response

Reply:
Does this work? https://github.com/strapi/strapi/commit/37c8c46162e3d2e01bbf90252c933bb2403fcd32

Reply:
There is the case where the id might not be requested:
fields: [name]

Even in that case, we would also need to fill "documentId" inside the list of fields to be selected

Only when fields is completely empty (all fields are selected) it shouldnt be necessary to select the documentID column